src/server/components/dashboard/controllers/controllers.ts

```diff
@@ -189,6 +189,34 @@ export async function listsEditController(req: Request, res: Response, next: Nex
   }
 }
 
+export async function listDeleteController(req: Request, res: Response) {
+  const listData = await findListDashboardData(req.params.listId);
+
+  res.render("dashboard/lists-delete", {
+    type: listData?.type,
+    country: listData?.country,
+    live: listData?.live,
+    listId: req.params.listId,
+  });
+}
+
+export async function listDeletePostController(req: Request, res: Response, next: NextFunction) {
+  const { listId } = req.params;
+  const list = await findListById(listId);
+  if (!req.user?.isAdministrator) {
+    const err = new HttpException(403, "403", "You is not authorised to execute this action.");
```

Comment:
Should be "You are not". 

I would probably write middleware which runs before the get handler and post handler and catch them there. I don't think admins should be able to reach the /delete page either.

```
listRouter.use("/:listId/delete", (req, res, next) => {
//
})
listRouter.get("/:listId/delete", listDeleteController);
listRouter.post("/:listId/delete", listDeletePostController);
```